tutorial/sample2/webgl-demo.js

```diff
@@ -1,66 +1,61 @@
-var canvas;
-var gl;
-var squareVerticesBuffer;
-var mvMatrix;
-var shaderProgram;
-var vertexPositionAttribute;
-var perspectiveMatrix;
+main();
 
 //
-// start
+// Start here
 //
-// Called when the canvas is created to get the ball rolling.
-// Figuratively, that is. There's nothing moving in this demo.
-//
-function start() {
-  canvas = document.getElementById("glcanvas");
+function main() {
+  const canvas = document.querySelector("#glcanvas");
+  const gl = canvas.getContext("webgl");
 
-  initWebGL(canvas);      // Initialize the GL context
+  // If we don't have a GL context, give up now
 
-  // Only continue if WebGL is available and working
+  if (!gl) {
+    alert("Unable to initialize WebGL. Your browser or machine may not support it.");
+    return;
+  }
 
-  if (gl) {
-    gl.clearColor(0.0, 0.0, 0.0, 1.0);  // Clear to black, fully opaque
-    gl.clearDepth(1.0);                 // Clear everything
-    gl.enable(gl.DEPTH_TEST);           // Enable depth testing
-    gl.depthFunc(gl.LEQUAL);            // Near things obscure far things
+  // Vertex shader program
 
-    // Initialize the shaders; this is where all the lighting for the
-    // vertices and so forth is established.
+  const vsSource = `
```

Comment:
Thank you, I always get the question of how those magic script tags worked. This is much simpler.